twitterscraper/main.py

```diff
@@ -98,14 +98,31 @@ def main():
             exit(-1)
 
         if args.all:
-            args.begindate = dt.date(2006,3,1)
+            args.begindate = dt.date(2006, 3, 1)
 
         if args.user:
-            tweets = query_tweets_from_user(user = args.query, limit = args.limit)
+            if args.javascript:
+                tweets = query_js.get_user_data(
+                    from_user=args.query, limit=args.limit,
+                    begindate=args.begindate, enddate=args.enddate,
+                    poolsize=args.poolsize, lang=args.lang
+                )['tweets']
+            else:
+                tweets = query.query_tweets_from_user(user=args.query, limit=args.limit)
+
         else:
-            tweets = query_tweets(query = args.query, limit = args.limit,
-                              begindate = args.begindate, enddate = args.enddate,
-                              poolsize = args.poolsize, lang = args.lang)
+            if args.javascript:
+                tweets = query_js.get_query_data(
+                    queries=[args.query], limit=args.limit,
+                    begindate=args.begindate, enddate=args.enddate,
+                    poolsize=args.poolsize, lang=args.lang
+                )['tweets']
```

Comment:
Since query_js.get_query_data() no longer returns a list of Tweet objects (like query.query_tweets() does) but returns a dictionary with the tweet-id as the key and a dump of the tweet object as the value, saving it to CSV no longer works.

Reply:
Agreed, converting this list of dicts to `Tweet` objects is necessary before merging.

Reply:
resolved in https://github.com/taspinar/twitterscraper/pull/302/commits/27ef71ca3e664ec0a79c2cfccbeb40bc8311891b